--- a/src/V3Hash.h
+++ b/src/V3Hash.h
@@ -46,8 +46,8 @@ class V3Hash final {
         : V3Hash{static_cast<uint32_t>(val)} {}
     explicit V3Hash(int64_t val)
         : V3Hash{static_cast<uint64_t>(val)} {}
-    explicit V3Hash(void* val)
-        : V3Hash{reinterpret_cast<uintptr_t>(val)} {}
+    explicit V3Hash(const void* val)
+        : V3Hash{static_cast<uint64_t>(reinterpret_cast<uintptr_t>(val))} {}
 
     // METHODS
     uint32_t value() const VL_MT_SAFE { return m_value; }
diff --git a/src/V3OrderMoveGraph.h b/src/V3OrderMoveGraph.h
index bab6ac6305..7d22320b8b 100644
--- a/src/V3OrderMoveGraph.h
+++ b/src/V3OrderMoveGraph.h
@@ -111,8 +111,8 @@ class OrderMoveDomScope final {
         struct Hash final {
             size_t operator()(const DomScopeMapKey& key) const {
                 // cppcheck-suppress unreadVariable  // cppcheck bug
-                V3Hash hash{reinterpret_cast<uint64_t>(key.m_domainp)};
-                hash += reinterpret_cast<uint64_t>(key.m_scopep);
+                V3Hash hash{key.m_domainp};
+                hash += key.m_scopep;
                 return hash.value();
             }
         };
